File: JAVA/S1_BASIC/L5_BASICRECURSION/reversearray.java

import java.util.ArrayList;
class recursion{
    void reverse(ArrayList<Integer>arr,int start,int end){
        if(start>=end)return;
        int t=arr.get(start);
        arr.set(start,arr.get(end));
        arr.set(end,t);
        reverse(arr,start+1,end-1);

    }
}
public class reversearray{
    public static void main(String[] args){
        ArrayList<Integer>arr=new ArrayList<>();
        arr.add(1);
        arr.add(2);
        arr.add(3);
        arr.add(4);
        arr.add(5);
        recursion r=new recursion();
        int n=arr.size();
        r.reverse(arr,0,n-1);
        System.out.print(arr);
    }
}

File: python/S1_BASIC/L5_BASICRECURSION/reversearray.py
class Recursion:
    def __init__(self,arr):
        self.arr=arr
    def reverse(self,arr,start,end):
        if start>=end:
            return
        arr[start],arr[end]=arr[end],arr[start]
        self.reverse(arr,start+1,end-1)
if __name__=="__main__":
    arr=list(map(int,input().split()))
    n=len(arr)
    R=Recursion(arr)
    R.reverse(arr,0,n-1)
    print(arr)
